cli/azd/pkg/executil/builder.go

```diff
@@ -0,0 +1,76 @@
+package executil
+
+import "context"
+
+type Builder struct {
+	commandRunnerFn RunCommandFn
+	runArgs         RunArgs
+}
+
+// Creates a new builder with commandRunnerFn, cmd and args
+func NewBuilder(cmd string, args ...string) *Builder {
+	runArgs := NewRunArgs(cmd, args...)
+
+	return &Builder{
+		runArgs: runArgs,
+	}
+}
+
+// Updates the commandRunnerFn that will used to execute commands
+func (b *Builder) WithCommandRunner(commandRunnerFn RunCommandFn) *Builder {
+	b.commandRunnerFn = commandRunnerFn
+	return b
+}
+
+// Updates the cmd that will be executed
+func (b *Builder) WithCmd(cmd string) *Builder {
+	b.runArgs.Cmd = cmd
+	return b
+}
+
+// Updates the command params that will get set
+func (b *Builder) WithParams(params ...string) *Builder {
+	b.runArgs.Args = params
+	return b
+}
```

Comment:
Seems like a better API would be `AppendParams()` for conditional parameter building

Reply:
Agreed - will update.